File: python/problem0010/solution.py
def primes(limit=None):
    found, x = [2], 3
    yield 2
    while limit is None or x < limit:
        for prim in found:
            if not x % prim:
                break
        else:
            yield x
            found.append(x)
        x += 2


print(sum(primes(2 * 10**6)))

File: go/problem0010/solution.go
// Find the sum of all primes below two million
package main

import (
	"eulerproject/utils/primes"
	"fmt"
)

func main() {
	limit := 2000000
	p := primes.NewPrimes()
	result := p.Get()
	fmt.Println("This can run a few seconds")
	for fp := p.Next(); fp < limit; fp = p.Next() {
		result += fp
	}
	fmt.Printf("Final result:\n%v\n", result)
}
